### PR TITLE
Add manual testing

### PR DESCRIPTION
**Pull request by ryan1604**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -491,7 +491,70 @@ shows you the welcome screen for the financial planner app
...
+
+Expected (Budget exists): Budget is displayed. Details of budget are shown in status message.
+
+Expected (Budget does not exist): No budget to display. Error details shown in the status message.
+
+### Displaying overview
+
+Test case: `overview`
+
+Expected: Displays overview of user's financials. Details of financials are shown in the status message.
```
Comment (wwweert123) [inline docs/DeveloperGuide.md:778]: LeGoat

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -491,7 +491,70 @@ shows you the welcome screen for the financial planner app
...
+
+Test case: `budget set /b 100`
+
+Expected: A monthly budget of 100 is set. Details of the budget are shown in the status message.
+
+Test case: `budget set /b`
+
+Expected: No budget is set. Error details shown in status message.
+
+Other incorrect set budget commands to try: `budget set`, `budget set /b x`, `...` (where x is negative)
```
Comment (wwweert123) [inline docs/DeveloperGuide.md:736]: what is ...

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -572,4 +635,39 @@ Again we can visualize these income using 3 different charts in separate command
...
+
+For incomes and expenses, the second column represent the amount, which is a `double`. To simulate a corrupted data, you 
+can change the number in the column to a string for example.
+
+Example of corrupted `data.txt` file in the third row:
+
+```
+I | 5000.0 | SALARY | 30 | false | 31/10/2023
+E | 50.0 | OTHERS | 0 | false
+I | sdf | OTHERS | 0 | false
```
Comment (wwweert123) [inline docs/DeveloperGuide.md:888]: Nice

**Review by wwweert123 — APPROVED** [review]
Victor wembanyama

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ryan1604) [reply docs/DeveloperGuide.md:736]: other incorrect commands for tester to come up with

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ryan1604) [reply docs/DeveloperGuide.md:778]: 1 billion net worth 💪

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (ryan1604) [reply docs/DeveloperGuide.md:888]: thanks!

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -572,4 +635,39 @@ Again we can visualize these income using 3 different charts in separate command
...
+I | 500.0 | OTHERS | 0 | false
+I | 5.0 | OTHERS | 0 | false
+E | 5.0 | OTHERS | 0 | false
+```
+
+The first column specifies the type of data being saved, and the subsequent columns contain the data to be saved. 
+For example, `I` and `E` represent `income` and `expense` respectively, and there are other types, such as `B` for `budget`. 
+
+For incomes and expenses, the second column represent the amount, which is a `double`. To simulate a corrupted data, you 
+can change the number in the column to a string for example.
```
Comment (NeoMinWei) [inline docs/DeveloperGuide.md:881]: string can be numbers also, be more specific

**Review by NeoMinWei — APPROVED** [review]
LGTM

**Pull request merged**